content/en/docs/languages/java/instrumentation.md

```diff
@@ -1144,72 +1144,237 @@ The metrics API defines a variety of instruments. Instruments record
+            // Default is 60000ms (60 seconds). Set to 10 seconds for demonstrative purposes only.
+            .setInterval(Duration.of(10000, ChronoUnit.MILLIS)).build())
+    .build();
+
+// Register MeterProvider with OpenTelemetry instance
+OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+    .setMeterProvider(sdkMeterProvider)
+    .build();
+```
 
-// Gets or creates a named meter instance
-Meter meter = openTelemetry.meterBuilder("instrumentation-library-name")
-        .setInstrumentationVersion("1.0.0")
-        .build();
+### Acquiring a Meter
+
+Anywhere in your application where you have manually instrumented code you can
+call `opentelemetry.meterBuilder(instrumentationScopeName)` to get or create a
+new meter instance using the builder pattern, or
+`opentelemetry.getMeter(instrumentationScopeName)` to get or create a meter
+based on just the instrument scope name.
+
+```java
+// Get or create a named meter instance with instrumentation version using builder
+Meter meter = openTelemetry.meterBuilder("dice-server")
+    .setInstrumentationVersion("0.1.0")
+    .build();
+
+// Get or create a named meter instance by name only
+Meter meter = openTelemetry.getMeter("dice-server");
+```
+
+Now that you have [meters](/docs/concepts/signals/metrics/#meter) initialized.
+you can create
+[metric instruments](/docs/concepts/signals/metrics/#metric-instruments).
+
+### Using Counters
+
+Counters can be used to measure a non-negative, increasing value.
+
+```java
+LongCounter counter = meter.counterBuilder("dice-lib.rolls.counter")
+    .setDescription("How many times the dice have been rolled.")
+    .setUnit("rolls")
+    .build();
+
+counter.add(1, attributes);
+```
+
+### Using UpDown Counters
+
+UpDown counters can increment and decrement, allowing you to observe a
+cumulative value that goes up or down.
+
+```java
+LongUpDownCounter counter = meter.upDownCounterBuilder("dice-lib.score")
+    .setDescription("Score from dice rolls.")
+    .setUnit("points")
+    .build();
+
+//...
+
+counter.add(10);
+
+//...
+
+counter.add(-20);
+```
+
+### Using Histograms
+
+Histograms are used to measure a distribution of values over time.
+
+```java
+LongHistogram histogram = meter.histogramBuilder("dice-lib.rolls")
+    .ofLongs() // Required to get a LongHistogram, default is DoubleHistogram
+    .setDescription("A distribution of the value of the rolls.")
+    .setUnit("points")
+    .build();
+
+histogram.record(7);
+```
+
+### Using Observable (Async) Counters
+
+Observable counters can be used to measure an additive, non-negative,
+monotonically increasing value.
+
+```java
+ObservableLongCounter counter = meter.counterBuilder("dice-lib.cache.size")
+    .buildWithCallback(measurement -> measurement.record(getCacheSize()));
+```
+
+### Using Observable (Async) UpDown Counters
+
+Observable UpDown counters can increment and decrement, allowing you to measure
+an additive, non-negative, non-monotonically increasing cumulative value.
```

Comment:
I guess there's no simpler way of describing this type of value?

Reply:
I used the same description that's currently used by [Go](http://localhost:1313/docs/languages/go/instrumentation/#using-observable-async-updown-counters) and [Javascript](http://localhost:1313/docs/languages/js/instrumentation/#using-observable-async-updown-counters), but they are the only languages with explicit descriptions so perhaps we could improve the clarity.

Some other ideas of ways we could phrase it:

- "Observable UpDown counters, capable of both increasing and decreasing at each measurement, offer a way to record cumulative, non-negative values that may not increase monotonically."
- "Observable UpDown counters support values that go up or down between measurements, making them suitable for tracking cumulative values that may not always rise."

One thing I've struggled with as a user, is the distinction between UpDown Counters vs Gauges. I think the primary distinction is the cumulative aspect of the UpDown compared to a Gauge being a snapshot where each measurement is independent of the previous value, so I've prioritized that aspect of the description, but open to other suggestions.

Reply:
IMO it's kind of inherently confusing. I'm fine with this wording.